core/src/main/java/io/opencensus/stats/View.java

```diff
@@ -14,18 +14,24 @@
 package io.opencensus.stats;
 
 import com.google.auto.value.AutoValue;
-import io.opencensus.common.Function;
+import io.opencensus.common.Duration;
 import java.util.ArrayList;
 import java.util.Collections;
 import java.util.List;
+import javax.annotation.Nullable;
 import javax.annotation.concurrent.Immutable;
 
 /**
  * A View specifies an aggregation and a set of tag keys. The aggregation will be broken
  * down by the unique set of matching tag values for each measure.
  */
 @Immutable
+@AutoValue
 public abstract class View {
+
+  View() {
+  }
+
   /**
    * Name of view. Must be unique.
    */
```

Comment:
I think we will need a way to look up data for a View by name, or at least a subset of the View fields.  For example, a request coming from the network shouldn't need to specify the view's description.  Even if we don't need the functionality in the first release, I think we should use the Name class here to avoid needing to change the signature after it's released.